ReactiveFeedback/StateMachine.swift

```diff
@@ -0,0 +1,73 @@
+import ReactiveSwift
+import Result
+
+/// An `EventBuffer` manages the deferred processing of recursive events.
+///
+/// When a given event is processed by a reducer, a new state would be computed together
+/// with new feedbacks, which could potentially yield a new event synchronously. Since
+/// the ReactiveSwift primitives do not support value recursion, the state machine would
+/// need to specifically cater for this scenario so as to be synchronous by default.
+private struct EventBuffer<Event> {
+    var isTransitioning: Bool
+    var events: ContiguousArray<Event>
+
+    /// Dequeue the first event in the queue, or complete the transitioning if the queue
+    /// is empty.
+    mutating func completeOrDequeue() -> Event? {
+        assert(isTransitioning)
+        guard events.isEmpty
+            else { return events.removeFirst() }
+        isTransitioning = false
+        return nil
+    }
+}
+
+internal enum StateMachine {
+    internal static func bootstrap<State, Event>(
+        state: SignalProducer<State, NoError>,
+        process: @escaping (Event) -> Void,
+        feedbacks: [Feedback<State, Event>],
+        lifetime: Lifetime
+    ) {
+        // Treat initialisation as an implicit event, and buffer any event yielded during
+        // the initialisation.
+        let buffer = Atomic(EventBuffer<Event>(isTransitioning: true, events: []))
+
+        state.startWithSignal { state, interruptHandle in
```

Comment:
don't we need to add `Disposable` to `lifetime`

Reply:
Did on line 37.

`startWithSignal` does not return a `Disposable`. What it does is:
1. make a `Signal`.
2. run the given `(Signal, Disposable) -> Void` closure.
3. start the deferred work.